lib/miq_automation_engine/engine/miq_ae_service.rb

```diff
@@ -318,6 +318,40 @@ def __find_instance_from_path(path)
       aec.ae_instances.detect { |i| instance.casecmp(i.name) == 0 }
     end
 
+    def with_acquired_lock(lock_name, max_retry_time = 60, retries = 10)
+      give_up_at = Time.zone.now + max_retry_time
+      retry_interval = max_retry_time / retries
+
+      while Time.zone.now < give_up_at
+        if MiqConcurrency::PGMutex.try_advisory_lock(lock_name)
+          begin
+            return yield
+          ensure
+            MiqConcurrency::PGMutex.release_advisory_lock(lock_name)
+          end
+          return
+        end
+        sleep(retry_interval)
```

Comment:
I know @mkanoor questioned it, but I think we should indeed be randomizing this. I'm thinking `retry_interval * (0.5 + rand(0.5))`.

It does make the `retries` parameter more of a minimum / guideline, though.